openshift-kube-apiserver/authorization/minimumkubeletversion/minimum_kubelet_version.go

```diff
@@ -0,0 +1,124 @@
+package minimumkubeletversion
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/blang/semver/v4"
+	nodelib "github.com/openshift/library-go/pkg/config/node"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/auth/nodeidentifier"
+)
+
+var (
+	minimumKubeletVersion *semver.Version
+	versionGotten         bool
+	versionSet            bool
+	versionLock           sync.Mutex
+)
+
+// GetMinimumKubeletVersion retrieves the set global minimum kubelet version in a safe way.
+// It ensures it is only retrieved once, and is set before it's retrieved.
+// It will panic on any error.
+// The global value should only be gotten through this function.
+func GetMinimumKubeletVersion() *semver.Version {
+	versionLock.Lock()
+	defer versionLock.Unlock()
+	if versionGotten {
+		panic("coding error: MinimumKubeletVersion already gotten")
+	}
+	if !versionSet {
+		panic("coding error: MinimumKubeletVersion not yet set")
+	}
+	versionGotten = true
+	return minimumKubeletVersion
+
+}
+
+// SetMinimumKubeletVersion sets the global minimum kubelet version in a safe way.
+// It ensures it is only set once, and the passed version is valid.
+// If will panic on any error.
+// The global value should only be set through this function.
+// Passing an empty string for version is valid.
+func SetMinimumKubeletVersion(version string) {
+	versionLock.Lock()
+	defer versionLock.Unlock()
+	if versionSet {
+		panic("coding error: MinimumKubeletVersion already set")
+	}
+	versionSet = true
+	if version == "" {
+		return
+	}
+	v := semver.MustParse(version)
+	minimumKubeletVersion = &v
+}
+
+type minimumKubeletVersionAuth struct {
+	nodeIdentifier nodeidentifier.NodeIdentifier
+	nodeInformer   corev1informers.NodeInformer
+	minVersion     *semver.Version
+	hasSyncedFunc  func() bool // added to make progress on a unit test, as the informer is spoofed and doesn't have anything to sync to
+}
+
+func NewMinimumKubeletVersion(nodeIdentifier nodeidentifier.NodeIdentifier,
```

Comment:
Should indicate on a godoc here that this requires exactly one call to SetMinimumKubeletVersion.